diffrax/_progress_bar.py

```diff
@@ -0,0 +1,186 @@
+        return jnp.array(0, dtype=jnp.int32)
+
+    def update(
+        self,
+        pb_id: IntScalarLike,
+        current_progress: FloatScalarLike,
+        previous_progress: FloatScalarLike,
+        current_step: IntScalarLike,
+    ) -> FloatScalarLike:
+        """
+        Args:
+            pb_id: progress bar id
+            current_progress: current progress in percent
+            previous_progress: previous progress in percent
+            current_step: current step number
+        """
+
+        def update_aux(progress):
+            # currently
+            # jax.debug.print("{progress:.2f}%", progress=progress)
+            # does not work, so we have to round the progress ourselves
+            jax.debug.print("{}%", np.round(progress, 1), ordered=True)
+            return progress
+
+        # we only print if the progress has increased by at least
+        # `minimum_increase` to avoid flooding the user
+        # with too many updates
+        return jax.lax.cond(
+            current_progress - previous_progress > self.minimum_increase,
+            update_aux,
+            lambda p: previous_progress,
+            current_progress,
+        )
+
+    def close(self, pb_id: IntScalarLike):
+        jax.debug.print("100%", ordered=True)
+
+
+class TqdmProgressBar(AbstractProgressBar):
+    refresh_rate: int
+
+    def __init__(self, refresh_rate: int = 20):
+        """
+        Args:
+            refresh_rate: number of steps between refreshes
+        """
+        self.refresh_rate = refresh_rate
+
+    def new(self) -> Array:
+        return io_callback(
+            progress_bar_manager.new, jax.ShapeDtypeStruct(tuple(), jnp.int32)
+        )
+
+    def update(
+        self,
+        pb_id: IntScalarLike,
+        current_progress: FloatScalarLike,
+        previous_progress: FloatScalarLike,
+        current_step: IntScalarLike,
+    ) -> FloatScalarLike:
+        """
+        Args:
+            pb_id: progress bar id
+            current_progress: current progress in percent
+            previous_progress: previous progress in percent
+            current_step: current step number
+        """
+
+        def update_aux(pb_id, progress):
+            io_callback(
+                progress_bar_manager.update,
+                progress,
+                progress,
+                pb_id,
+                ordered=True,
+            )
+            return progress
+
+        # here we update every `refresh_rate` steps in order
+        # to limit `io_callback` expensive calls
+        return jax.lax.cond(
+            current_step % self.refresh_rate == 0,
+            update_aux,
+            lambda _, progress: progress,
+            pb_id,
+            current_progress,
+        )
+
+    def close(self, pb_id: IntScalarLike):
+        """
+        Args:
+            pb_id: progress bar id
+        """
+        io_callback(progress_bar_manager.close, None, pb_id, ordered=True)
+
+
+class TqdmProgressBarManager:
```

Comment:
Nit: can you start this with an underscore to indicate that it's private to this module?